tests/unit/test_messageDialog.py

```diff
@@ -0,0 +1,903 @@
+			print(mocked_messageBoxCounter.mock_calls)
+			mocked_showModal.assert_called_once()
+			mocked_messageBoxCounter.__iadd__.assert_called_once()
+			mocked_messageBoxCounter.__isub__.assert_called_once()
+
+		# raise Exception
+
+
+class Test_MessageDialog_EventHandlers(MDTestBase):
+	def test_onShowEventDefaultFocus(self):
+		"""Test that _onShowEvent correctly focuses the default focus."""
+		self.dialog.addOkButton().addCancelButton(defaultFocus=True)
+		evt = wx.ShowEvent(self.dialog.GetId(), True)
+		with patch.object(wx.Window, "SetFocus") as mocked_setFocus:
+			self.dialog._onShowEvent(evt)
+			mocked_setFocus.assert_called_once()
+
+	def test_onCloseEventNonVetoable(self):
+		evt = wx.CloseEvent(wx.wxEVT_CLOSE_WINDOW, self.dialog.GetId())
+		"""Test that a non-vetoable close event is executed."""
+		evt.SetCanVeto(False)
+		self.dialog._instances.append(self.dialog)
+		with (
+			patch.object(wx.Dialog, "Destroy") as mocked_destroy,
+			patch.object(
+				self.dialog,
+				"_executeCommand",
+				wraps=self.dialog._executeCommand,
+			) as mocked_executeCommand,
+		):
+			self.dialog._onCloseEvent(evt)
+			mocked_destroy.assert_called_once()
+			mocked_executeCommand.assert_called_once_with(ANY, _canCallClose=False)
+			self.assertNotIn(self.dialog, MessageDialog._instances)
+
+	def test_onCloseEventNoFallbackAction(self):
+		"""Test that a vetoable call to close is vetoed if there is no fallback action."""
+		self.dialog.addYesNoButtons()
+		self.dialog.SetEscapeId(EscapeCode.NO_FALLBACK)
+		evt = wx.CloseEvent(wx.wxEVT_CLOSE_WINDOW, self.dialog.GetId())
+		MessageDialog._instances.append(self.dialog)
+		with (
+			patch.object(wx.Dialog, "DestroyLater") as mocked_destroyLater,
+			patch.object(
+				self.dialog,
+				"_executeCommand",
+			) as mocked_executeCommand,
+		):
+			self.dialog._onCloseEvent(evt)
+			mocked_destroyLater.assert_not_called()
+			mocked_executeCommand.assert_not_called()
+			self.assertTrue(evt.GetVeto())
+			self.assertIn(self.dialog, MessageDialog._instances)
+
+	def test_onCloseEventFallbackAction(self):
+		"""Test that _onCloseEvent works properly when there is an there is a fallback action."""
+		self.dialog.addOkCancelButtons()
+		evt = wx.CloseEvent(wx.wxEVT_CLOSE_WINDOW, self.dialog.GetId())
+		MessageDialog._instances.append(self.dialog)
+		with (
+			patch.object(wx.Dialog, "DestroyLater") as mocked_destroyLater,
+			patch.object(
+				self.dialog,
+				"_executeCommand",
+				wraps=self.dialog._executeCommand,
+			) as mocked_executeCommand,
+		):
+			self.dialog._onCloseEvent(evt)
+			mocked_destroyLater.assert_called_once()
+			mocked_executeCommand.assert_called_once_with(ANY, _canCallClose=False)
+			self.assertNotIn(self.dialog, MessageDialog._instances)
+
+	@parameterized.expand(
+		(
+			("closableCanCallClose", True, True, True),
+			("ClosableCannotCallClose", True, False, False),
+			("UnclosableCanCallClose", False, True, False),
+			("UnclosableCannotCallClose", False, False, False),
```

Comment:
is there a way to parametize these with argument names to enhance readability? it's okay if not, just would be handy

e.g.
```suggestion
			("closableCanCallClose", closesDialog=True, canCallClose=True, expectedCloseCalled=True),
			("ClosableCannotCallClose", True, False, False),
			("UnclosableCanCallClose", False, True, False),
			("UnclosableCannotCallClose", False, False, False),
```

Reply:
Yep, I can just create `NamedTuple`s for these parameter lists.